ShaiRandom/Collections/ProbabilityTable.cs

```diff
@@ -17,106 +18,152 @@ public class ProbabilityTable<TItem>
+        /// </param>
+        public ProbabilityTable(IEnumerable<(TItem item, double weight)> items) : this(new MizuchiRandom(), items.ToArray())
         {
         }
 
         /// <summary>
         /// Constructs a ProbabilityTable that will use the given (item, weight) pairs and an unseeded MizuchiRandom.
         /// </summary>
-        /// <param name="items">An IReadOnlyList of pairs, where the first item of a pair is the TItem to be potentially returned, and the second item is the weight for how much to favor returning that item.</param>
-        public ProbabilityTable(IReadOnlyList<(TItem item, double weight)> items) : this(new MizuchiRandom(), items)
+        /// <param name="items">
+        /// An IReadOnlyList of pairs, where the first item of a pair is the TItem to be potentially returned,
+        /// and the second item is the weight for how much to favor returning that item.  The list will NOT be copied,
+        /// and must not be modified after it is passed to this function.
+        /// </param>
+        public ProbabilityTable(ref IReadOnlyList<(TItem item, double weight)> items) : this(new MizuchiRandom(), items)
         {
         }
+
+
         /// <summary>
         /// Constructs a ProbabilityTable that will use the given (item, weight) pairs and the given IEnhancedRandom.
         /// </summary>
         /// <param name="random">Any IEnhancedRandom, such as a <see cref="TrimRandom"/> or <see cref="LaserRandom"/>.</param>
-        /// <param name="items">An IReadOnlyList of pairs, where the first item of a pair is the TItem to be potentially returned, and the second item is the weight for how much to favor returning that item.</param>
-        public ProbabilityTable(IEnhancedRandom random, IReadOnlyList<(TItem item, double weight)> items)
+        /// <param name="items">
+        /// A list of pairs, where the first item of a pair is the TItem to be potentially returned, and the
+        /// second item is the weight for how much to favor returning that item.
+        /// </param>
+        public ProbabilityTable(IEnhancedRandom random, IEnumerable<(TItem item, double weight)> items) : this(random, items.ToArray())
+        {
+        }
+
+        /// <summary>
+        /// Constructs a ProbabilityTable that will use the given (item, weight) pairs and the given IEnhancedRandom.
+        /// </summary>
+        /// <param name="random">Any IEnhancedRandom, such as a <see cref="TrimRandom"/> or <see cref="LaserRandom"/>.</param>
+        /// <param name="items">
+        /// An IReadOnlyList of pairs, where the first item of a pair is the TItem to be potentially returned, and the
+        /// second item is the weight for how much to favor returning that item.  The list will NOT be copied,
+        /// and must not be modified after it is passed to this function.
+        /// </param>
+        public ProbabilityTable(IEnhancedRandom random, ref IReadOnlyList<(TItem item, double weight)> items) : this(random, items)
```

Comment:
Does this need to use `ref items` in the call to `this()`?

Reply:
Not the way I wrote it; the private constructor _doesn't_ take by-ref, so that's what this will resolve to.